processor.go

```diff
@@ -18,26 +20,50 @@ type Payload struct {
 	Success interface{} `json:"success,omitempty"`
 }
 
-func Start(h interface{}) {
+type MockLambda struct {
+	api     func(h func(ctx context.Context, request events.APIGatewayProxyRequest) (events.APIGatewayProxyResponse, error)) Response
+	token   func(h func(request events.APIGatewayCustomAuthorizerRequest) (events.APIGatewayCustomAuthorizerResponse, error)) Response
+	request func(h func(request events.APIGatewayCustomAuthorizerRequestTypeRequest) (events.APIGatewayCustomAuthorizerResponse, error)) Response
+	sqs     func(h func(ctx context.Context, request events.SQSEvent) error) Response
+}
+
+func (ml *MockLambda) start(h interface{}) error {
 	response := Response{}
 
-	authorizer := os.Getenv("IS_LAMBDA_AUTHORIZER")
-	requestAuthorizer := os.Getenv("IS_LAMBDA_REQUEST_AUTHORIZER")
-	tokenAuthorizer := os.Getenv("IS_LAMBDA_TOKEN_AUTHORIZER")
+	types := reflect.TypeOf(h)
+	inputCount := types.NumIn()
 
-	// Most common - probably
-	if authorizer != "true" {
-		response = api(h.(func(ctx context.Context, request events.APIGatewayProxyRequest) (events.APIGatewayProxyResponse, error)))
-	}
+	inputTypes := make([]string, inputCount)
 
-	if authorizer == "true" && tokenAuthorizer == "true" {
-		response = token(h.(func(request events.APIGatewayCustomAuthorizerRequest) (events.APIGatewayCustomAuthorizerResponse, error)))
+	for i := 0; i < inputCount; i++ {
+		inputTypes[i] = types.In(i).String()
 	}
 
-	if authorizer == "true" && requestAuthorizer == "true" {
-		response = request(h.(func(request events.APIGatewayCustomAuthorizerRequestTypeRequest) (events.APIGatewayCustomAuthorizerResponse, error)))
+	fmt.Println(inputTypes)
```

Comment:
We should remove this due to how the go implementation handles stdout on the javascript side.